### PR TITLE
improve range

### PR DESCRIPTION
**Pull request by zyxxoo**

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java [hunk 1]
```diff
@@ -308,9 +308,7 @@ public void rollback() {
             int tablePrefixLength = startKey.size();
 
             List<Pair<byte[], byte[]>> result = regions.stream()
-                      .filter((region) -> {
-                             return !region.getStartKey().isEmpty();
-                       }).map((region) -> {
+                        .map((region) -> {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java]: can we replace the code with for-loop style? and then adjust the alignment

Comment (zyxxoo) [reply src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java]: ok

File: src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java [hunk 2]
```diff
@@ -301,62 +298,58 @@ public void rollback() {
...
+                ByteString regionEndKey = region.getEndKey();
+                if (regionEndKey != null && !regionEndKey.isEmpty()) {
+                    Key regionEndRowKey = Key.toRawKey(regionEndKey);
+                    if (realEndKey.compareTo(regionEndRowKey) >= 0) {
+                        realEndKey = regionEndRowKey;
+                    }
+                }
+
+                pairs.add(Pair.of(originKey(tablePrefixLength,
+                                  realStartKey.getBytes()),
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java]: align with realStartKey logically?

File: src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java [hunk 3]
```diff
@@ -301,62 +298,58 @@ public void rollback() {
...
+                                  originKey(tablePrefixLength,
+                                  realEndKey.getBytes())));
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java]: ditto

File: src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java [hunk 4]
```diff
@@ -301,62 +298,58 @@ public void rollback() {
...
+            if (CollectionUtils.isNotEmpty(pairs)) {
+                return pairs;
             }
 
-            return ImmutableList.of(Pair.of(new byte[0],
-                                            originKey(tablePrefixLength,
-                                                      endKey.toByteArray())));
+            pairs.add(Pair.of(new byte[0],
+                              originKey(tablePrefixLength,
+                              endKey.toByteArray())));
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java]: ditto

File: src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java [hunk 5]
```diff
@@ -301,62 +298,58 @@ public void rollback() {
...
                                                                 endKey);
+
+            List<Pair<byte[], byte[]>> pairs = new ArrayList<>();
             if (regions == null) {
-                return Collections.emptyList();
+                return pairs;
             }
 
             int tablePrefixLength = startKey.size();
+            for (TiRegion region : regions) {
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/backend/store/tikv/TikvStdSessions.java:308]: looks better

**Review by javeme — APPROVED** [review]

**Pull request merged**